lib/src/chart/radar_chart/radar_chart_painter.dart

```diff
@@ -0,0 +1,310 @@
+import 'package:flutter/material.dart';
+
+import '../../../fl_chart.dart';
+
+class RadarChartPainter extends BaseChartPainter<RadarChartData>
+    with TouchHandler<RadarTouchResponse> {
+  final Paint _borderPaint, _backgroundPaint, _gridPaint, _tickPaint;
+  final Paint _graphPaint, _graphBorderPaint, _graphPointPaint;
+  final TextPainter _ticksTextPaint, _titleTextPaint;
+
+  List<RadarDataSetsPosition> dataSetsPosition;
+
+  RadarChartPainter(
+    RadarChartData data,
+    RadarChartData targetData,
+    Function(TouchHandler) touchHandler, {
+    double textScale = 1,
+  })  : _backgroundPaint = Paint()
+          ..color = data.fillColor
+          ..style = PaintingStyle.fill
+          ..isAntiAlias = true,
+        _borderPaint = Paint()
+          ..color = data?.chartBorderData?.color ?? Colors.black
+          ..strokeWidth = data?.chartBorderData?.width ?? 2
+          ..style = PaintingStyle.stroke
+          ..isAntiAlias = true,
+        _gridPaint = Paint()
+          ..color = data?.gridBorderData?.color ?? Colors.black
+          ..strokeWidth = data?.gridBorderData?.width ?? 2
+          ..style = PaintingStyle.stroke
+          ..isAntiAlias = true,
+        _tickPaint = Paint()
+          ..color = data?.tickBorderData?.color ?? Colors.black
+          ..strokeWidth = data?.tickBorderData?.width ?? 2
+          ..style = PaintingStyle.stroke
+          ..isAntiAlias = true,
+        _graphPaint = Paint(),
+        _graphBorderPaint = Paint(),
+        _graphPointPaint = Paint(),
+        _ticksTextPaint = TextPainter(),
+        _titleTextPaint = TextPainter(),
+        super(data, targetData, textScale: textScale) {
+    touchHandler(this);
+  }
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    super.paint(canvas, size);
+
+    if (data.dataSets.isEmpty) return;
+
+    dataSetsPosition = _calculateDataSetsPosition(size);
+
+    drawTicks(size, canvas);
+    drawGrids(size, canvas);
+    drawTitles(size, canvas);
+    drawDataSets(size, canvas);
+  }
+
+  void drawTicks(Size size, Canvas canvas) {
+    final centerX = radarCenterX(size);
+    final centerY = radarCenterY(size);
+    final centerOffset = Offset(centerX, centerY);
+
+    /// controls Radar chart size
+    final radius = radarRadius(size);
+
+    //draw radar background
+    canvas.drawCircle(centerOffset, radius, _backgroundPaint);
+    //draw radar border
+    canvas.drawCircle(centerOffset, radius, _borderPaint);
+
+    final dataSetMaxValue = data.maxEntry.value;
+    final dataSetMinValue = data.minEntry.value;
+    final tickSpace = (dataSetMaxValue - dataSetMinValue) / data.tickCount;
+
+    final ticks = <double>[];
+
+    for (var tick = dataSetMinValue; tick <= dataSetMaxValue; tick = tick + tickSpace)
+      ticks.add(tick);
+
+    final tickDistance = radius / (ticks.length);
+
+    final ticksStyle = data.ticksTextStyle ?? const TextStyle(fontSize: 10, color: Colors.black);
```

Comment:
We must assume that it is a non-null value and if you want you can set this default value in the RadarChartData constructor (just like other default values). It helps us to migrate to the dart null-safety feature smoothly.